### DISCUSSION TITLE
Can field settings / conditions such as masking or hiding fields also apply to the response data?

### DISCUSSION THREAD

**Discussion by Elprede** [directus/directus#16983] [Feature Requests]
Hi Directus team,

Just wanted to ask if its possible to also apply the masking / hidden settings of field values on the JSON response data?

I know from the admin configurations we can set masking and hiding of fields only to the collections detail view only.

Use case example:
I have a collection (eg called Issues) and in this collection I have a field called "Confidential" (boolean), when I set this field to True, another field gets Masked or Hidden.

What occurs:
The admin details view on the item gets set to Hidden / Mask when the Confidential field on that item = True however on the API call, the field that was masked / hidden is still shown which kind of defeats the purpose.

What should happen:
The api acknowledges that the condition is set and when Confidential = True, the field is hidden (or if its Masked, maybe the value is set to "MASKED")

Just a suggestion to help with this awesome project :)

**Comment by br41nslug** [comment]
Field conditions and interface options currently only work on the fronted, they change how the frontend behaves but does not directly affect the data returned on the API or it's validation rules.
Conditionally making data confidential sounds very fragile to me, that would mean if a toggle is accidentally flipped suddenly everyone would have access to that previously confidential data. Personally i'd split restricted and not restricted data in separate tables and make sure only the users/roles that need it have access to the restricted one. You could add the confidential toggle and using a Flow either save it to the regular table or to the restricted table based on the toggle.
As for masking values that should really be covered by permissions, masked fields really should not have the permissions to know that fields even exists.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
